### DISCUSSION TITLE
Permissions - allow access to all objects for a site

### DISCUSSION THREAD

**Discussion by travisjohnsonga** [netbox-community/netbox#13406] [General]
Is there a way in permissions to allow all objects associated with a particular site, without creating a permission for each object type?

**Comment by mathewbishop** [comment]
Hey @travisjohnsonga, I'm thinking you _may_ be able to do something like:

![image](https://github.com/netbox-community/netbox/assets/39043451/faeef3da-64ef-4c79-a778-e88afc61d52d)

<br/>

Creating a constraint based on `site__name` or maybe `site__name__slug`, something along those lines. 

In the Objects section, you'd want to select each object type that you want the permission to display, and it should then only display the objects that have a relation to that particular site name, due to the constraint. Of course, the permission may fail to save if you select an object type that does not have a relation to the `Site` object, as in that case the constraint will be looking for something that isn't there. So the objects selected have to have to be objects that are able to belong to a given site.

Hope this helps!
